harmonica/tests/test_gradient_boosted_eqs.py

```diff
@@ -0,0 +1,263 @@
+    # Compute expected required memory
+    sources_p_window = (int(window_size / spacing) + 1) ** 2
+    data_p_window = (int(window_size / spacing) + 1) ** 2
+    expected_required_memory = data_p_window * sources_p_window * itemsize
+    # Estimate required memory
+    required_memory = EquivalentSourcesGB.estimate_required_memory(
+        coordinates, window_size=window_size, dtype=dtype
+    )
+    assert required_memory == expected_required_memory
+
+
+# -----------------------------------------------------------------------
+# Test the fitting and predictions of gradient-boosted equivalent sources
+# -----------------------------------------------------------------------
+
+
+@pytest.mark.use_numba
+@pytest.mark.parametrize("weights", [None, np.ones((8, 8))])
+def test_gb_eqs_small_data(coordinates_small, data_small, weights):
+    """
+    Check predictions against synthetic data using few data points for speed
+    """
+    # The interpolation should be good enought on the data points
+    # Gradient-boosted equivalent sources don't perform well on small data, so
+    # we will check if the error is no larger than 1mGal
+    # (sample data ranges from approximately -7mGal to 7mGal)
+    eqs = EquivalentSourcesGB(depth=1e3, damping=None, window_size=1e3, random_state=42)
+    eqs.fit(coordinates_small, data_small, weights=weights)
+    npt.assert_allclose(
+        data_small, eqs.predict(coordinates_small), atol=0.05 * vd.maxabs(data_small)
+    )
+
+
+@run_only_with_numba
+def test_gradient_boosted_eqs_single_window(region, points, masses, coordinates, data):
+    """
+    Test GB eq-sources with a single window that covers the whole region
+    """
+    # The interpolation should be perfect on the data points
+    eqs = EquivalentSourcesGB(window_size=region[1] - region[0])
+    eqs.fit(coordinates, data)
+    npt.assert_allclose(data, eqs.predict(coordinates), rtol=1e-5)
+    # Gridding onto a denser grid should be reasonably accurate when compared
+    # to synthetic values
+    grid = vd.grid_coordinates(region, shape=(60, 60), extra_coords=0)
+    true = point_mass_gravity(grid, points, masses, field="g_z")
+    npt.assert_allclose(true, eqs.predict(grid), rtol=1e-3)
+
+
+@run_only_with_numba
+def test_gradient_boosted_eqs_predictions(region, points, masses, coordinates, data):
+    """
+    Test GB eq-sources predictions
+    """
+    # The interpolation should be sufficiently accurate on the data points
+    eqs = EquivalentSourcesGB(window_size=1e3, depth=1e3, damping=None, random_state=42)
+    eqs.fit(coordinates, data)
+    npt.assert_allclose(data, eqs.predict(coordinates), atol=0.02 * vd.maxabs(data))
+
+    # Gridding onto a denser grid should be reasonably accurate when compared
+    # to synthetic values
+    grid = vd.grid_coordinates(region, shape=(60, 60), extra_coords=0)
+    true = point_mass_gravity(grid, points, masses, field="g_z")
+    npt.assert_allclose(true, eqs.predict(grid), atol=0.02 * vd.maxabs(true))
```

Comment:
With adding a comment that the error tolerance is 2% of the maximum data. And probably best to set `rtol=0` since it defaults to `1e-5`.

Reply:
> Makes perfect sense and it says so in the docstring which is perfect. How about just shuffle since the function is already _create_windows?

I used `shuffle` at first, but the linter warned me that `shuffle` was taken by the import of `sklearn.utils.shuffle` function. I agree with renaming the argument, so I would need to import the function with an alias.